### ISSUE TITLE
lldpd: add support for SNMP agentx/master

### ISSUE THREAD

**Issue by jubork** [openwrt/openwrt#13878]
### Describe the bug

Current LLDPD package (file name: lldpd_1.0.17-1_mipsel_24kc.ipk) does not support
SNMP agentx/master interface (the -x command line option).
Attempt to activate SNMP interface results in :

root@openwrt:~# /usr/sbin/lldpd -x
SNMP support is not built-in
Usage:   lldpd [OPTIONS ...]
Version: lldpd 1.0.17
[.....]

SNMP interface for LLDPD is useful for 
network monitoring agents which utilize
SNMP protocol a lot.

It should be easy to add SNMP support : 
*) update package's configure/make files to enable SNMP build option
*) add snmpd/libsnmp to package dependencies
  

### OpenWrt version

r23497-6637af95aa

### OpenWrt target/subtarget

ramips/mt7621

### Device

Xiaomi Mi Router 3G

### Image kind

Official downloaded image

### Steps to reproduce

root@openwrt:~# /usr/sbin/lldpd -x
SNMP support is not built-in
Usage:   lldpd [OPTIONS ...]
Version: lldpd 1.0.17

### Actual behaviour

lldpd daemon reports that it does not support the '-x' option.

### Expected behaviour

lldpd daemon should accept the '-x' option,like so:

 lldpd -x -d
2023-11-05T20:07:39 [INFO/main] protocol LLDP enabled
2023-11-05T20:07:39 [INFO/main] protocol CDPv1 disabled
2023-11-05T20:07:39 [INFO/main] protocol CDPv2 disabled
2023-11-05T20:07:39 [INFO/main] protocol SONMP disabled
2023-11-05T20:07:39 [INFO/main] protocol EDP disabled
2023-11-05T20:07:39 [INFO/main] protocol FDP disabled
2023-11-05T20:07:39 [INFO/event] libevent 2.1.12-stable initialized with epoll method
2023-11-05T20:07:39 [INFO/snmp] enable SNMP subagent
2023-11-05T20:07:39 [INFO/libsnmp] NET-SNMP version 5.9.3 AgentX subagent connected
2023-11-05T20:07:39 [INFO/lldpctl] lldpd should resume operations

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by Managor** [comment]
I would like this too. Having LLDP and SNMP work together would be critical for using LanTopoLog to map out the local network topology.

**Comment by diepeterpan** [comment]
I need this too. I had to compile my own for 3 different Architectures as LibreNMS works nicely with LLDP and SNMP. Please, can we get this package?

**Comment by iotola** [comment]
Yes I would like it for the same reason using LibreNMS and getting the topology of LLDP via SNMP.
